cli/internal/plan/plan.go

```diff
@@ -0,0 +1,162 @@
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/fatih/color"
+	"github.com/mitchellh/cli"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/vercel/turborepo/cli/internal/config"
+	"github.com/vercel/turborepo/cli/internal/context"
+	"github.com/vercel/turborepo/cli/internal/scm"
+	"github.com/vercel/turborepo/cli/internal/scope"
+	"github.com/vercel/turborepo/cli/internal/ui"
+)
+
+type PlanCommand struct {
+	config *config.Config
+	ui     cli.Ui
+}
+
+func NewCmd(config *config.Config, UI cli.Ui) *PlanCommand {
+	return &PlanCommand{
+		config: config,
+		ui:     UI,
+	}
+}
+
+func (pc *PlanCommand) Help() string {
+	cmd := pc.getCmd()
+	return cmd.UsageString()
+}
+
+func (pc *PlanCommand) Run(args []string) int {
+	cmd := pc.getCmd()
+	cmd.SetArgs(args)
+	err := cmd.Execute()
+	if err != nil {
+		pc.config.Logger.Error("error", err)
+		pc.ui.Error(fmt.Sprintf("%s%s", ui.ERROR_PREFIX, color.RedString(" %v", err)))
+		return 1
+	}
+	return 0
+}
+
+func (pc *PlanCommand) Synopsis() string {
+	cmd := pc.getCmd()
+	return cmd.Short
+}
+
+type planOpts struct {
+	Scope                []string
+	GlobalDeps           []string
+	Since                string
+	Ignore               []string
+	Cwd                  string
+	IncludeDepdendencies bool
+	NoDependents         bool
+	OutputJSON           bool
+}
+
+func (po *planOpts) ScopeOpts() *scope.Opts {
+
+	// includeDependencies := false
+	// if po.IncludeDepdendencies != nil {
+	// 	includeDependencies = po.IncludeDepdendencies
+	// } else if po.Since != "" && len(po.Scope) != 0 {
+	// 	includeDependencies = true
+	// }
+	return &scope.Opts{
+		IncludeDependencies: po.IncludeDepdendencies,
+		IncludeDependents:   !po.NoDependents,
+		Patterns:            po.Scope,
+		Since:               po.Since,
+		Cwd:                 po.Cwd,
+		IgnorePatterns:      po.Ignore,
+		GlobalDepPatterns:   po.GlobalDeps,
+	}
+}
+
+func (pc *PlanCommand) getCmd() *cobra.Command {
+	opts := &planOpts{}
+	cmd := &cobra.Command{
+		Use:   "turbo plan",
+		Short: "Display packages affected by 'turbo run'",
+		Long:  "Display the packages that would be affected by 'turbo run' with similar options",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// match the logic from run opts. If --since and --scope are specified,
+			// assume --include-dependencies. If the user has specified --include-dependencies,
+			// use the user's value.
+			depsWasSet := cmd.Flags().Changed("include-dependencies")
+			if !depsWasSet && opts.Since != "" && len(opts.Scope) != 0 {
+				opts.IncludeDepdendencies = true
+			}
+			return plan(pc.config, pc.ui, opts)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringSliceVar(&opts.Scope, "scope", nil, "Specify package(s) to act as entry points for task\nexecution. Supports globs.")
+	flags.StringSliceVar(&opts.GlobalDeps, "global-deps", nil, "Specify glob of global filesystem dependencies to\nbe hashed. Useful for .env and files in the root\ndirectory. Can be specified multiple times.")
```

Comment:
Would this override the global dependencies defined in the`turbo.json` pipeline file?

Reply:
Yes